### DISCUSSION TITLE
If you use Docker, you can set this variable from a file by using OOO_OOO_FILE?

### DISCUSSION THREAD

**Discussion by logankim0913** [firefly-iii/firefly-iii#6749] [Firefly III] [answered]
Hello! As I was fiddling with .env file in firefly iii, I saw comments like the title around variable declaration lines that need to be configured manually. What do these comments really mean?

**Comment by JC5** [comment, marked as answer]
If you gotta ask... 😉 

You must have noticed how your database passwords etc. are essentially plain text inside a `.env` file or `docker-compose.yml`. There are 'secret managers' like Docker Swarm and Hashicorp Vault that "inject" the secret directly into the container by mounting a file somewhere in the container. For example, `/var/run/mysql_password` would be a text file that contains the... MySQL password. It can only be read by that specific container and the file doesn't really exist except as a memory-only secret.

To read such variables, the Firefly III container is capable of reading the file linked to in `FOOBAR_FILE` and store the content as a variable in `FOOBAR`. It does so during container startup.

**Reply by logankim0913** [reply]
Thank you!
